### PR TITLE
feat: enhance input validation utilities

### PR DESCRIPTION
**Pull request by tzervas**
## Summary
This PR enhances the validation utilities with structured results, better error handling, and comprehensive testing.

### 1. ValidationResult Class
- Structured result type for validation operations
- Contains validation status, errors, and warnings
- Clear initialization with type safety

### 2. Enhanced Placeholder Validation
- Better handling of invalid argument types
- Improved warning messages for missing/unused arguments
- Structured validation results

### 3. Improved Tag Validation
- Specific warnings for different tag format issues
- Better handling of non-string tags
- Improved error messages for validation failures

### 4. Comprehensive Test Suite
- Unit tests for ValidationResult class
- Parametrized tests for validation scenarios
- Tests for specific warning messages
- Edge case handling tests

## Benefits
- Better error handling and reporting
- More informative validation messages
- Consistent validation result structure
- Improved type safety
- Comprehensive test coverage

## Test Plan
- Added tests for all validation utilities
- Verified error handling for invalid inputs
- Checked warning generation for edge cases
- Tested validation result structure

### CODE REVIEW

File: src/warp_content_processor/utils/validation.py [hunk 1]
```diff
@@ -1,12 +1,32 @@
...
+
+    def __init__(
+        self,
+        is_valid: bool = True,
+        errors: Optional[List[str]] = None,
+        warnings: Optional[List[str]] = None,
+    ):
+        self.is_valid = is_valid
+        self.errors = errors or []
+        self.warnings = warnings or []
```
[the remark this answers is not in the document]
Comment (tzervas) [reply src/warp_content_processor/utils/validation.py:24]: @sourcery-ai issue
